src/YesSql.Provider.Common/SqlBuilder.cs

```diff
@@ -198,6 +207,18 @@ public virtual string ToSqlString()
 
                 sb.Append("SELECT ");
 
+                if (_distinct)
+                {
+                    sb.Append("DISTINCT ");
+
+                    // This hacky bit needs a better solution
+                    if (_order != null && _order[0] != _dialect.QuoteForColumnName("Id"))
```

Comment:
So actually the `_distinct` should be a string that contains the value as a parameter, optionally, like `""` to mean `DISTINCT` and `some thing` to mean `DISTINCT(some thing)`, and that something should be built using the dialect from outside the builder. Because here the `SqlBuilder` should not have to know about a column name (`[Id]`).

Reply:
I don't think I want to parameterize it yet.  It isn't needed to solve the current issue I'm trying to solve.  It would just create new problems to solve.